### ISSUE TITLE
PHPunit plugin FAILED even if test is success

### ISSUE THREAD

**Issue by davidnjuko** [Block8/PHPCI#310]
Hi,
I've installed PHPCI today with PHPUnit support and when i'm building my project i have this message :

```
PLUGIN STATUS: SUCCESS!
RUNNING PLUGIN: php_unit
PHPUnit 3.7.32 by Sebastian Bergmann.

Configuration read from /module/Application/tests/phpunit.xml

...............................................................  63 / 209 ( 30%)
............................................................... 126 / 209 ( 60%)
............................................................... 189 / 209 ( 90%)
....................

Time: 605 ms, Memory: 21.75Mb

OK (209 tests, 510 assertions)
PHPUnit 4.0.12 by Sebastian Bergmann.

PLUGIN STATUS: FAILED
RUNNING PLUGIN: mysql
PLUGIN STATUS: SUCCESS!
RUNNING PLUGIN: email
1 emails sent
0 emails failed to send
PLUGIN STATUS: SUCCESS!
```

Any ideay why Plugin failed even if unit test is success ?

**Comment by jralison** [comment]
@davidnjuko I am facing the same problem with my instance of phpci. My phpunit tests pass on my computer but they fail in phpci without any message. Would you tell me what error had you found in your composer installation?

I already checked my php log files, my nginx log files and I have found nothing. In which log may I find something wrong with composer configuration?

If, by chance, someone else can give me any clue about what may be wrong with my files I am sending below my phpci.yml content and the Build Log content I get when the build completes.

Thank you in advance.

### phpci.yml
```
setup:
  composer:
    action: "install"
    prefer_dist: true
  env:
    YII_ENV: "stage"

test:
  php_unit:
    directory: "app/tests/unit"
    args: "--bootstrap app/tests/bootstrap.php --no-configuration"

(...)
```

### Build Log
```
Cloning into '/var/www/phpci/PHPCI/build/10_db392'...

6004f0965e197e8c806efa4a628e1f7fed3aae2a
Working copy created: /
RUNNING PLUGIN: composer
Using --prefer-dist flag
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
  - Installing symfony/yaml (v3.0.3)
    Loading from cache

  - Installing symfony/polyfill-mbstring (v1.1.1)
    Loading from cache

  - Installing symfony/console (v3.0.3)
    Loading from cache

  - Installing symfony/filesystem (v3.0.3)
    Loading from cache

  - Installing symfony/config (v3.0.3)
    Loading from cache

  - Installing robmorgan/phinx (v0.5.3)
    Loading from cache

  - Installing swiftmailer/swiftmailer (v5.4.1)
    Loading from cache

  - Installing tecnickcom/tcpdf (6.2.12)
    Loading from cache

  - Installing yiisoft/yii (1.1.17)
    Loading from cache

  - Installing webmozart/assert (1.1.0)
    Loading from cache

  - Installing phpdocumentor/reflection-common (1.0)
    Loading from cache

  - Installing phpdocumentor/type-resolver (0.2)
    Loading from cache

  - Installing phpdocumentor/reflection-docblock (3.1.0)
    Loading from cache

  - Installing phpunit/php-token-stream (1.4.8)
    Loading from cache

  - Installing sebastian/version (2.0.0)
    Loading from cache

  - Installing sebastian/resource-operations (1.0.0)
    Loading from cache

  - Installing sebastian/recursion-context (1.0.2)
    Loading from cache

  - Installing sebastian/object-enumerator (1.0.0)
    Loading from cache

  - Installing sebastian/global-state (1.1.1)
    Loading from cache

  - Installing sebastian/exporter (1.2.2)
    Loading from cache

  - Installing sebastian/environment (1.3.7)
    Loading from cache

  - Installing sebastian/diff (1.4.1)
    Loading from cache

  - Installing sebastian/comparator (1.2.0)
    Loading from cache

  - Installing phpunit/php-text-template (1.2.1)
    Loading from cache

  - Installing doctrine/instantiator (1.0.5)
    Loading from cache

  - Installing phpunit/phpunit-mock-objects (3.2.3)
    Loading from cache

  - Installing phpunit/php-timer (1.0.8)
    Loading from cache

  - Installing phpunit/php-file-iterator (1.4.1)
    Loading from cache

  - Installing sebastian/code-unit-reverse-lookup (1.0.0)
    Loading from cache

  - Installing phpunit/php-code-coverage (4.0.1)
    Loading from cache

  - Installing phpspec/prophecy (v1.6.1)
    Loading from cache

  - Installing myclabs/deep-copy (1.5.1)
    Loading from cache

  - Installing phpunit/phpunit (5.5.0)
    Loading from cache

symfony/console suggests installing psr/log (For using the console logger)
symfony/console suggests installing symfony/event-dispatcher ()
symfony/console suggests installing symfony/process ()
sebastian/global-state suggests installing ext-uopz (*)
phpunit/php-code-coverage suggests installing ext-xdebug (>=2.4.0)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
Generating autoload files

PLUGIN: SUCCESS
RUNNING PLUGIN: env
PLUGIN: SUCCESS
RUNNING PLUGIN: php_unit
PLUGIN: FAILED
BUILD FAILED
RUNNING PLUGIN: slack_notify
PLUGIN: SUCCESS
Removing Build.
```

**Comment by eduardocruz** [comment]
PS: The issue description should be "PHPunit plugin FAILED even if test is success ... IN MY COMPUTER" because the reason the build is failing is because PHPUnit is actually failing. 

I was having the same problem. The way I found the problem, was, I stopped the build after it copied the code and I tried running phpunit from the build folder. Then I realized that the reson for that is that I missed environment variables that were present in my local code folder (that's why it worked in my local folder) and these variables didn't exist in the build folder. Then I used the ENV plugin to add the missing api key my project needed to run the tests. I my case it was the Bugsnag API key, which was missing and would make the phpunit tests break. 

Then after that I realized I I'm using Laravel and I wasn't running migrations, which would create the database tables I needed. Since the build was using a blank database. 

That said, my suggestion is that you run you project from the build folder and try to find out, because the reason is probably the configuration/environment in your computer is not exactly the same as where the build is running.
